### PR TITLE
Add target job to restart test

### LINKED ISSUE #3330: RRTMGP introduces errors in restarted simulations
Testing with this configuration:
```
                    test_dict = Dict(
                        "test_dycore_consistency" => true, # We will add NaNs to the cache, just to make sure
                        "check_nan_every" => 3,
                        "log_progress" => false,
                        "moist" => "equil",
                        "precip_model" => "0M",
                        "config" => configuration,
                        "topography" => "NoWarp",
                        "turbconv" => "diagnostic_edmfx",
                        "dt" => "1secs",
                        "bubble" => bubble,
                        "viscous_sponge" => true,
                        "rayleigh_sponge" => true,
                        "insolation" => "timevarying",
                        "rad" => "clearsky",
                        "dt_rad" => "1secs",
                        "surface_setup" => "DefaultMoninObukhov",
                        # "implicit_diffusion" => true,
                        "call_cloud_diagnostics_per_stage" => true,  # Needed to ensure that cloud variables are computed
                        "t_end" => "3secs",
                        "dt_save_state_to_disk" => "1secs",
                        "enable_diagnostics" => false,
                        "output_dir" => joinpath(output_loc, job_id),
                    )
```
Here are some of the differences (there's more)
```
Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669424f0 <= 1.1920929f-5
     Context: name = integrator.u.c.uₕ.components.data.1
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669423f0 <= 1.1920929f-5
     Context: name = integrator.u.c.uₕ.components.data.2
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 1.6163609f-5 <= 1.1920929f-5
     Context: name = integrator.u.f.u₃.components.data.1
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669424f0 <= 1.1920929f-5
     Context: name = integrator.p.precomputed.ᶜu.components.data.1
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669423f0 <= 1.1920929f-5
     Context: name = integrator.p.precomputed.ᶜu.components.data.2
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669424f0 <= 1.1920929f-5
     Context: name = integrator.p.precomputed.ᶜuʲs.1.components.data.1
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669423f0 <= 1.1920929f-5
     Context: name = integrator.p.precomputed.ᶜuʲs.1.components.data.2
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669424f0 <= 1.1920929f-5
     Context: name = integrator.p.precomputed.ᶜu⁰.components.data.1
              what = reading and simulating
              config = sphere equil 0M NoWarp

Test Failed at /home/sbozzolo/repos/ClimaAtmos.jl/test/restart.jl:42
  Expression: maximum(error) <= 100 * eps(eltype(arr1))
   Evaluated: 0.3669423f0 <= 1.1920929f-5
     Context: name = integrator.p.precomputed.ᶜu⁰.components.data.2
              what = reading and simulating
              config = sphere equil 0M NoWarp
```

### PR DESCRIPTION
**Pull request by Sbozzolo**
This commit adds a new test in the restart file that uses a yml file as a base for a test. The job was picked to be a target job, but with reduced t_end (`amip_target_diagedmf.ylm`).

I would like to run everything together, but compilation time is too long. I changed the test to take a command-line argument to choose between running one comprehensive test (default) or many tests and added both on buildkite.

In working on this, I found a problem with our macOS runner (it is running in emulated mode, making it much slower). This PR includes a commit to fix that. I also removed some unused dependencies from the test environment.

I also found the issue with radiation allskywithclear, so I think that #3330 can be closed

What's left is mainly MPI

Closes #3330

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -28,15 +28,13 @@ jobs:
           - ubuntu-latest
           - macOS-latest
           - windows-latest
-        arch:
-          - x64
```
Comment (glwagner) [inline .github/workflows/ci.yml:32]: This is also a problem for ClimaOcean GitHub @navidcy

Comment (glwagner) [reply .github/workflows/ci.yml:32]: What was the original point of specifying architecture ?

**Comment by Sbozzolo** [comment]
Here is a successful build: https://buildkite.com/clima/climaatmos-ci/builds/21021

File: test/restart.jl [hunk 2]
```diff
@@ -27,6 +29,12 @@ ClimaComms.init(comms_ctx)
 # 2. A simulation, saved to a previous checkpoint, and read back and evolved to
 #   the same time is identical (up to some tolerance)
 # 3. ClimaAtmos can automatically detect restarts
+#
+# This test checks a variety of configurations and spends a long time compiling
+# functions. For this reason, the test takes an optional command-line argument
+# to only many tests instead of comprehensive test.
+
+const MANYTESTS = length(ARGS) >= 1
```
Comment (charleskawczynski) [inline test/restart.jl:37]: Can we make and extract this as a kwarg instead?

Comment (Sbozzolo) [reply test/restart.jl:37]: Would this be fine:
```julia
const MANYTESTS = "--manytest" in ARGS
```
?

Comment (charleskawczynski) [reply test/restart.jl:37]: Yeah, I think that's fine

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -429,6 +429,22 @@ steps:
           slurm_gpus: 1
           slurm_mem: 16G
 
+      - label: ":computer: test restart"
+        command: >
+          julia --color=yes --project=examples test/restart.jl MANYTESTS
```
Comment (charleskawczynski) [inline .buildkite/pipeline.yml:434]:
```suggestion
          julia --color=yes --project=examples test/restart.jl --MANYTESTS true
```
Just a reminder: we'll need to update this too

Comment (Sbozzolo) [reply .buildkite/pipeline.yml:434]: Yes, thank you

File: .github/workflows/ci.yml (hunk 1 above)
Comment (navidcy) [reply .github/workflows/ci.yml:32]: Yeah, arch is there because the default `.yml` file included it.

**Review by charleskawczynski — APPROVED** [review]

**Pull request merged**